lib/utils.js

```diff
@@ -157,31 +159,38 @@ utils.createMapOfType = (obj, Type) => {
   if (!obj) return result;
 
   Object.keys(obj).forEach(key => {
+    // eslint-disable-next-line security/detect-object-injection
     result[key] = new Type(obj[key]);
   });
 
   return result;
 };
 
 utils.getMapKeyOfType = (obj, key, Type) => {
-  if (!obj) return null;
-  if (!obj[key]) return null;
-  return new Type(obj[key]);
+  if (typeof key !== 'string' || !obj) return null;
+  const v = obj[String(key)];
+  if (v === undefined) return null;
+  return Type ? new Type(v) : v;
+};
+
+utils.getMapKey = (obj, key) => {
+  return utils.getMapKeyOfType(obj, key);
 };
 
 utils.addExtensions = (obj) => {
   obj.prototype.extensions = function () {
     const result = {};
     Object.keys(this._json).forEach(key => {
```

Comment:
same as with others, we could make `this._json` a Map

Reply:
But then users will have to get data via the `get ()` method and not via the array index, and for me this is a breaking change. If we decided to use Map here, then we should also adjust whole methods, which return plain JS Object, to Map. Or probably I don't understand your suggestion :D

Reply:
you change it to a map only for `forEach` but in 193 still return pure json